File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());
        string ans="";
        string f=strs[0];
        int n =strs.size();
        string last=strs[n-1];
        for(int i=0;i<min(f.size(), last.size());i++){
            if(f[i]!=last[i])return ans;
            ans+=f[i];
        }
        // unordered_map<string, int> mpp;
        // string word;
        return ans;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        string=""
        length=len(strs)
        
        for i in range(min(len(strs[0]), len(strs[-1]))):
            if strs[0][i]!=strs[-1][i]:
                return string
            string+=strs[0][i]
            
        return string